trieste/models/gpflow/architectures.py

```diff
@@ -0,0 +1,228 @@
+    likelihood_variance: Optional[float] = None,
+) -> VGP:
+    empirical_mean, empirical_variance, _ = _get_data_stats(data)
+
+    if classification:
+        if noise_free:
+            empirical_variance = 100.0
+        else:
+            empirical_variance = 1.0
+        model_likelihood = gpflow.likelihoods.Bernoulli()
+    else:
+        if likelihood_variance is None:
+            noise_variance = 0.1 * empirical_variance
+        else:
+            noise_variance = likelihood_variance
+        model_likelihood = gpflow.likelihoods.Gaussian(noise_variance)
+
+    kernel = _get_kernel(
+        empirical_variance, search_space, kernel_priors, classification, noise_free
+    )
+    mean = gpflow.mean_functions.Constant(empirical_mean)
+
+    model = gpflow.models.VGP(data.astuple(), kernel, model_likelihood, mean_function=mean)
+
+    gpflow.set_trainable(model.likelihood, (not noise_free))
+    if classification and noise_free:
+        gpflow.set_trainable(model.kernel.variance, False)
+
+    return model
+
+
+def build_svgp(
+    data: Dataset,
+    search_space: Box,
+    classification: bool = False,
+    kernel_priors: bool = True,
+    noise_free: bool = False,
+    likelihood_variance: Optional[float] = None,
+    num_inducing_points: Optional[int] = None,
+    trainable_inducing_points: bool = False,
+) -> SVGP:
+    empirical_mean, empirical_variance, num_data_points = _get_data_stats(data)
+
+    if classification:
+        if noise_free:
+            empirical_variance = 100.0
+        else:
+            empirical_variance = 1.0
+        model_likelihood = gpflow.likelihoods.Bernoulli()
+    else:
+        if likelihood_variance is None:
+            noise_variance = 0.1 * empirical_variance
+        else:
+            noise_variance = likelihood_variance
+        model_likelihood = gpflow.likelihoods.Gaussian(noise_variance)
+
+    kernel = _get_kernel(
+        empirical_variance, search_space, kernel_priors, classification, noise_free
+    )
+    mean = gpflow.mean_functions.Constant(empirical_mean)
+
+    inducing_points = _get_inducing_points(search_space, num_inducing_points)
+
+    model = SVGP(
+        kernel,
+        model_likelihood,
+        inducing_points,
+        mean_function=mean,
+        num_data=num_data_points,
+    )
+
+    gpflow.set_trainable(model.likelihood, (not noise_free))
+    gpflow.set_trainable(model.inducing_variable, trainable_inducing_points)
+    if classification:
+        gpflow.set_trainable(model.kernel.variance, False)
+
+    return model
+
+
+def _get_data_stats(data: Dataset) -> tuple[TensorType, TensorType, int]:
+    empirical_variance = tf.math.reduce_variance(data.observations)
+    empirical_mean = tf.math.reduce_mean(data.observations)
+    num_data_points = len(data.observations)
+
+    return empirical_mean, empirical_variance, num_data_points
+
+
+def _get_kernel(
+    variance: TensorType,
+    search_space: Box,
+    priors: bool,
+    classification: bool = False,
```

Comment:
maybe it would be clearer to have Booleans like `add_prior_to_lengthscale`, `add_prior_to_variance`